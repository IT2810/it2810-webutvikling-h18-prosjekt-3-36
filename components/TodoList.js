import React from 'react';
import {
  ScrollView,
} from 'react-native';

import TodoElement from "./TodoElement";

export default class TodoList extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      todos: [{ id: 1, text: "kjøp melk", date: "2018-10-04", category: "work", completed: false }, { id: 2, text: "gjør webdev og balbajsklfjkalsdfjkla sdjklfja skldfj klasjdfkl jaklsdfjkla jsdklfj askldjfkl jasdklfj klasjdfk jaskldfjklasj klfjaskld fjklasdfjkl ", date: "2018-10-05", category: "work", completed: true }
    ]};
  }

  render() {
    let todos = this.state.todos.map((todo) => {
      return (
        <TodoElement 
        key={todo["id"]}
        text={todo["text"]}
        date={todo["date"]}
        completed={todo["completed"]} />
      )
    });

    return (
      <ScrollView>{todos}</ScrollView>
    );
  }
}