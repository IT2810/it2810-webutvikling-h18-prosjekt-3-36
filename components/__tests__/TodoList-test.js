import React from "react";
import ShallowRenderer from 'react-test-renderer/shallow';
import TodoList from "../TodoList";
import { Provider } from 'react-redux'
import appState from '../../reducers/Reducer'
import { createStore } from 'redux'

const store = createStore(appState)


it("TodoList Matches snapshot.", () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Provider store={store}><TodoList /></Provider>)
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
});
