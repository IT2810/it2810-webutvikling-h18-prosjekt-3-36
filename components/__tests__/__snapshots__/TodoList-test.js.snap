// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList Matches snapshot. 1`] = `
<Component
  style={
    Object {
      "height": "25%",
    }
  }
>
  <ScrollViewMock />
</Component>
`;
